serval-agent/src/api/storage.rs

```diff
@@ -16,7 +16,12 @@ pub async fn get_blob(
     // Yeah, I don't like this.
     let state = state.lock().await;
 
-    match state.storage.get_stream(&blob_addr).await {
+    let Some(storage) = state.storage.as_ref() else {
+        // todo: in this case, we should proxy this request to another node that is advertising the serval_storage role
+        return (StatusCode::SERVICE_UNAVAILABLE, "Storage not available").into_response();
```

Comment:
I'd love to just be able to do something like:

````rust
return proxy_request_to_role("serval_storage", format!("/blob/{blob_addr}"));
````

It'd be even cooler if there was an easy way to get the current request we're in the middle of serving and pass it along to that hypothetical `proxy_request_to_role` function.